### ISSUE TITLE
Checkpoints don't use schema for formatting

### ISSUE THREAD

**Issue by micimize** [datahq/dataflows#59]
It looks like the `checkpoint` formatters uses [static unconfigurable serializers](https://github.com/datahq/dataflows/blob/master/dataflows/processors/dumpers/file_formats.py#L50).

This creates serious issues - even if the `datetime` formatter was timezone aware, it would then be adding timezone data to naive datetimes

**Comment by akariv** [comment]
I agree this is an issue - timezones are not handled properly.
Thanks for reporting and I'll fix this asap.

**Comment by akariv** [comment]
Fixed in v0.0.38

**Comment by micimize** [comment]
Wait, even if you replaced the mechanism for checkpointing, don't all the dumpers still lose timezone info? Can we not leverage the `datapackage.save`?

As an aside, the util I came up for for (I think) the equivalent of `stream/unstream` used `jsonpickle`:
```python
import tempfile, os
import jsonpickle as json


class LDJsonTemporaryFile(object):
    """
    Use a line delimited tempfile for caching and reading back an arbitrary stream of data.
    * will automatically delete the temporary file on close and __exit__
    * Used by `partition`
    """

    def __init__(self):
        self.file = tempfile.TemporaryFile(mode="r+")

    def append(self, obj):
        self.file.write(json.dumps(obj) + "\n")

    def __iter__(self):
        self.file.seek(0)
        for line in self.file:
            yield json.loads(line)

    def __enter__(self):
        return self

    def close(self):
        self.file.close()
        del self.file

    def __exit__(self, *exc_info):
        self.close()
```
